pkg/cmd/dev/doctor.go

```diff
@@ -11,22 +11,64 @@
 package main
 
 import (
-	"errors"
+	"context"
+	"io/ioutil"
 	"log"
-	"os/exec"
+	"os"
+	osexec "os/exec"
 	"path/filepath"
 	"runtime"
 	"strconv"
 	"strings"
 
+	"github.com/cockroachdb/cockroach/pkg/cmd/dev/io/exec"
+	"github.com/cockroachdb/errors"
 	"github.com/spf13/cobra"
 )
 
+const devStatusFile = "bin/.dev-status"
+
+// devStatusVersion is the current "version" of the status checks performed by
+// `dev doctor``. Increasing it will force doctor to be re-run before other dev
+// commands can be run.
+const devStatusVersion = 1
+
+func checkDoctorStatus(ctx context.Context, ex *exec.Exec) error {
```

Comment:
Nit: I would prefer to see a way to override/ignore this check via e.g. an environment variable, but I guess people can just write a number to `bin/.dev-status` if they want to slide under the radar here so it's fine?